mealie/repos/repository_cookbooks.py

```diff
@@ -0,0 +1,65 @@
+from collections.abc import Iterable
+
+from fastapi import HTTPException, status
+from pydantic import UUID4
+from slugify import slugify
+from sqlalchemy.exc import IntegrityError
+
+from mealie.db.models.household.cookbook import CookBook
+from mealie.repos.repository_generic import HouseholdRepositoryGeneric
+from mealie.schema.cookbook.cookbook import ReadCookBook, SaveCookBook
+from mealie.schema.response.responses import ErrorResponse
+
+
+class RepositoryCookbooks(HouseholdRepositoryGeneric[ReadCookBook, CookBook]):
+    def create(self, data: SaveCookBook | dict) -> ReadCookBook:
+        if isinstance(data, dict):
+            data = SaveCookBook(**data)
+        data.slug = slugify(data.name)
+
+        max_retries = 10
+        for i in range(max_retries):
+            try:
+                return super().create(data)
+            except IntegrityError:
+                self.session.rollback()
+                data.slug = slugify(f"{data.name} ({i+1})")
+
+        raise  # raise the last IntegrityError
+
+    def create_many(self, data: Iterable[ReadCookBook | dict]) -> list[ReadCookBook]:
+        return [self.create(entry) for entry in data]
+
+    def update(self, match_value: str | int | UUID4, data: SaveCookBook | dict) -> ReadCookBook:
+        if isinstance(data, dict):
+            data = SaveCookBook(**data)
+
+        new_slug = slugify(data.name)
+        if not (data.slug and data.slug.startswith(new_slug)):
+            data.slug = slugify(data.name)
```

Comment:
[question] Because this is `startswith` rather than an exact match, could that lead to problems?

Reply:
I was going to answer why this logic makes sense, but then I realized using regex makes way more sense here, so I updated it.

We can still run into weird behavior if the cookbook name actually contains a number in the title (e.g. if you rename a cookbook from "My Cookbook" to "My Cookbook 1" the slug won't update) but this is a tiny edge case, and it doesn't really cause any problems anyway besides the slug being slightly different than the name